src/main_menu.c

```diff
@@ -1,15 +1,1955 @@
+}
+
+void DrawMainMenuWindowBorder(const struct WindowTemplate *template, u16 baseTileNum)
+{
+    u16 r9 = 1 + baseTileNum;
+    u16 r10 = 2 + baseTileNum;
+    u16 sp18 = 3 + baseTileNum;
+    u16 spC = 5 + baseTileNum;
+    u16 sp10 = 6 + baseTileNum;
+    u16 sp14 = 7 + baseTileNum;
+    u16 r6 = 8 + baseTileNum;
+    
+    FillBgTilemapBufferRect(template->priority, baseTileNum, template->tilemapLeft - 1, template->tilemapTop - 1, 1, 1, 2);
+    FillBgTilemapBufferRect(template->priority, r9, template->tilemapLeft, template->tilemapTop - 1, template->width, 1, 2);
+    FillBgTilemapBufferRect(template->priority, r10, template->tilemapLeft + template->width, template->tilemapTop - 1, 1, 1, 2);
+    FillBgTilemapBufferRect(template->priority, sp18, template->tilemapLeft - 1, template->tilemapTop, 1, template->height, 2);
+    FillBgTilemapBufferRect(template->priority, spC, template->tilemapLeft + template->width, template->tilemapTop, 1, template->height, 2);
+    FillBgTilemapBufferRect(template->priority, sp10, template->tilemapLeft - 1, template->tilemapTop + template->height, 1, 1, 2);
+    FillBgTilemapBufferRect(template->priority, sp14, template->tilemapLeft, template->tilemapTop + template->height, template->width, 1, 2);
+    FillBgTilemapBufferRect(template->priority, r6, template->tilemapLeft + template->width, template->tilemapTop + template->height, 1, 1, 2);
+    CopyBgTilemapBufferToVram(template->priority);
+}
+
+void sub_8032250(const struct WindowTemplate *template)
+{
+    FillBgTilemapBufferRect(template->priority, 0, template->tilemapLeft - 1, template->tilemapTop - 1, template->tilemapLeft + template->width + 1, template->tilemapTop + template->height + 1, 2);
+    CopyBgTilemapBufferToVram(template->priority);
+}
+
+void sub_8032298(u8 a, u8 b, u8 c, u8 d, u8 e, u8 unused)
+{
+    FillBgTilemapBufferRect(a, 0, b + 0xFF, c + 0xFF, d + 2, e + 2, 2);
+}
+
+void sub_80322E0(u8 windowId, u8 a)
+{
+    CallWindowFunction(windowId, sub_8032298);
+    FillWindowPixelBuffer(windowId, 17);
+    ClearWindowTilemap(windowId);
+    if (a == 1)
+        CopyWindowToVram(windowId, 3);
+}
+
+void sub_8032318(u8 a)
+{
+    u8 fontAttribute = GetFontAttribute(1, 6);
+    u8 fontAttribute2 = GetFontAttribute(1, 0);
+    u8 fontAttribute3 = GetFontAttribute(1, 1);
+    u8 windowAttribute = GetWindowAttribute(a, 3);
+    u8 windowAttribute2 = GetWindowAttribute(a, 4);
+    
+    FillWindowPixelRect(a, fontAttribute, 0, 0, fontAttribute2 * windowAttribute, fontAttribute3 * windowAttribute2);
+    CopyWindowToVram(a, 2);
+}
+
+void sub_80323A0(struct TextSubPrinter *printer, u16 a)
+{
+    if (*(printer->current_text_offset - 2) == 8 && gUnknown_02022D04 == 0)
+    {
+        gUnknown_02022D04 = 1;
+        CreateTask(sub_8030A70, 0);
+    }
+}
+
+void sub_80323CC(u8 a, u8 b, u16 c, u16 d, u8 e, u8 f)
+{
+    struct WindowTemplate sp;
+    struct WindowTemplate sp2;
+    
+    sp = sub_8198A50(0, a + 1, b + 1, 5, 4, f, d);
+    sp2 = sp;
```

Comment:
This probably isn't needed and is just a result of the weird return-by-value nature of the function